### PR TITLE
Blog app: add authorization rules

### PR DESCRIPTION
**Pull request by Progress2002**
### Hello Reviewer,

This is @Progress2002  and @shafiuyushawu, and this milestone fulfills the below requirements:

### Project requirements
- Install CanCanCan in your project.
- Add a `role` column to the users table. Remember to use a migration for this.
- A user can delete a post if it is theirs or if they have an admin role (column `role` has value `"admin"`). Use CanCanCan for this authorization.
    - For that you need to implement the post deleting functionality. Add the "Delete" button to the view and make sure that only authorized users can see it.
- A user can delete a comment if it is theirs or if they have an admin role (column `role` has value `"admin"`). Use CanCanCan for this authorization.
    - For that you need to implement the comment deleting functionality. Add the "Delete" button to the view and make sure that only authorized users can see it.

### CODE REVIEW

File: app/controllers/posts_controller.rb [hunk 1]
```diff
@@ -26,12 +26,24 @@ def show
     @comments = Comment.where(post_id: params[:id])
   end
 
+  def destroy
+    @user = User.find(params[:user_id])
+    @post = @user.posts.find(params[:id])
+    authorize! :destroy, @post
+    @post.destroy
+    redirect_to user_posts_path(current_user, @post), notice: 'Post was successfully destroyed.'
+  end
```
Comment (elisha2kyakpo1) [inline app/controllers/posts_controller.rb]: - [ ] Great job adding the ability class to implement the authorization of users, but there is work to be done. I created two accounts using different browsers, I was able to delete a post and a comment created by the first user which should not be the case. I also noticed that one user (first) was creating posts this behavior should be fixed as well. Kindly make changes and submit it for another review. You can use `load_and_authorize_resource` in the Post controller for __CanCanCan__ to work properly.

**Review by elisha2kyakpo1 — CHANGES_REQUESTED** [review]
Hi 👋 @Progress2002 and @shafiuyushawu, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights

✔️ The PR title and description are good and professional.
✔️ No linter errors.
✔️ Descriptive readme file.
✔️ The CanCanCan gem and the ability class were set up👏

## Required Changes ♻️
_Check the comments under the review._

- [x] The partner with few commits to make changes until commits are equal.

## Optional suggestions

- [ ] The Postgres database should be set up during the project creation to avoid errors.

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/controllers/posts_controller.rb (hunk 1 above)
Comment (Progress2002) [reply app/controllers/posts_controller.rb]: Hello @elisha2kyakpo1 , thanks for the observation. The project has been updated to fix the issue.

**Review by addod19 — APPROVED** [review]
### Hi @Progress2002 and @shafiuyushawu, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

### Highlights
- [ ] Nice job addressing the comment from the previous review :rocket: 


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**